File: src/modules/organization/domain/dto/organization_update_dto.py
from ninja import Schema
from pydantic import Field
from src.modules.address.domain.dto.address_update_dto import AddressUpdateDTO
from src.shared.domain.enums.status_enum import Status


class OrganizationUpdateDTO(Schema):
    name: str | None = None
    trade_name: str | None = None
    email: str | None = Field(
        None,
        pattern=r"^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$",
    )
    status: Status | None = None
    address: AddressUpdateDTO | None = None

File: src/modules/organization/domain/dto/organization-update.dto.ts
import { AddressUpdateDto } from '@/shared/domain/dto/address-update.dto';
import { EStatus } from '@/shared/domain/enums/status.enum';

export interface OrganizationUpdateDto {
  name: string;
  tradeName: string;
  email: string;
  status: EStatus;
  address: AddressUpdateDto;
  avatarId?: number;
}
